server/py/services/api/initial_data.py

```diff
@@ -993,6 +1002,46 @@ def _create_project_summaries(db, db_session):
     db._upsert(db_session, project_summaries, ignore=True)
 
 
+def _init_system_id(db_session: sqlalchemy.orm.Session):
+    # Initializes a system id for MLRun deployment.
+    # The system id is first checked in the database. If it does not exist, the function checks for a configured id
+    # (from environment variables or config), and if neither is found, a new random one is generated and stored.
+    db = framework.db.sqldb.db.SQLDB()
+
+    # check if a system id already exists in the database
+    system_id = db.get_system_id(db_session)
+
+    if system_id is not None:
+        logger.info("Existing system id found in the database", system_id=system_id)
+        return
+
+    # check if the system id is configured via environment variables or configmap
+    system_id = _get_configured_system_id()
+
+    if system_id is not None:
+        logger.info("Using configured system id", system_id=system_id)
+        db.create_system_id(db_session, system_id)
+        return
+
+    # if no system id is found, generate a new one
+    system_id_str = _generate_system_id()
+    db.create_system_id(db_session, system_id_str)
+
+
+def _get_configured_system_id() -> typing.Optional[str]:
+    system_id = os.environ.get("SYSTEM_ID")
+    if system_id:
+        return system_id
+    return mlrun.mlconf.system_id or None
+
+
+def _generate_system_id() -> str:
+    # Generate a random 32-bit unsigned integer and encode it as base64 string without padding
+    random_int = os.urandom(4)
```

Comment:
os.urandom(4) doesn't generate `a random 32-bit unsigned integer `, it generates some a random mix of bytes
```
os.urandom(4)
Out[16]: b'?7\xe2x'
```

Reply:
@Yacouby but should they be random bytes? Either the requirement + task should be rephrased or logic should be aligned with the original idea of using integer.

Reply:
@liranbg raised a good point with possible `+` in the encoding, I believe that's the reason why originally it was only unsigned int - base64 encoding of strings 0000 to 9999 never includes non-alphabetic symbols because their binary representations map only to alphanumeric characters in the Base64 alphabet.